### PR TITLE
[PNUDEV-14] [PNUDEV-4] Implemented security, users and writer accounts

### LINKED ISSUE #4: Implement Security

### LINKED ISSUE #14: Implement users and writers accounts
1. Each writer should have an account with short description.
2. Implement CRUD for accounts.

### PR DESCRIPTION
**Pull request by Igor-Polatajko**
Login page:
image

Admin panel (Admin view):
image

Admin menu:
image

Admin panel (Writer view):
image

Writer menu:
image

Edit account page:
image

Update password:
image

Admin panel user management:
image

Create new user form:
image

New user was created
image

Deactivate account:
image

### CODE REVIEW

File: src/main/java/com/pnu/dev/shpaltaif/aop/ModelAttributeSetter.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    @ModelAttribute
+    public void setCategories(HttpServletRequest request, Model model) {
+
+        if (!StringUtils.startsWith(request.getRequestURI(), "/admin")) {
+            List<Category> categories = categoryService.findAll();
+            model.addAttribute("categories", categories);
+        }
+
+    }
```
Comment (volodymyrvynnyk) [inline src/main/java/com/pnu/dev/shpaltaif/aop/ModelAttributeSetter.java]: Is it called by each request?

File: src/main/java/com/pnu/dev/shpaltaif/controller/PublicAccountController.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    @PostMapping("/update")
+    public String update(@AuthenticationPrincipal User user, @Validated PublicAccountDto publicAccountDto,
+                         RedirectAttributes redirectAttributes) {
+
+        Long publicAccountId = user.getPublicAccount().getId();
+        publicAccountService.update(publicAccountDto, publicAccountId);
+
+        authSessionSynchronizer.refreshPrincipalInAuthSession(user.getId());
+
+        redirectAttributes.addFlashAttribute(FLASH_MESSAGE_SUCCESS, "Анаунт було успішно оновлено");
```
Comment (volodymyrvynnyk) [inline src/main/java/com/pnu/dev/shpaltaif/controller/PublicAccountController.java]: Probably, you were tired)

File: src/main/java/com/pnu/dev/shpaltaif/service/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+    @Transactional
+    @Override
+    public void create(CreateUserDto createUserDto) {
+
+        if (!StringUtils.equals(createUserDto.getPassword(), createUserDto.getRepeatedPassword())) {
+            throw new ServiceAdminException("Паролі не співпадають!");
+        }
+
+        if (userRepository.existsByUsername(createUserDto.getPassword())) {
```
Comment (volodymyrvynnyk) [inline src/main/java/com/pnu/dev/shpaltaif/service/UserServiceImpl.java]: getUsername instead of getPassword

**Review by volodymyrvynnyk — CHANGES_REQUESTED** [review]
Looks great! Few comments, and attach UI's screenshots, please

File: src/main/java/com/pnu/dev/shpaltaif/aop/ModelAttributeSetter.java (hunk 1 above)
Comment (Igor-Polatajko) [reply src/main/java/com/pnu/dev/shpaltaif/aop/ModelAttributeSetter.java]: Yes, it does.
The idea here is next: we need to render sidebar with categories on each page, except pages related to admin panel.
So this logic just updates the model with categories attribute for each request that is not related to admin panel.

File: src/main/java/com/pnu/dev/shpaltaif/controller/PublicAccountController.java (hunk 2 above)
Comment (Igor-Polatajko) [reply src/main/java/com/pnu/dev/shpaltaif/controller/PublicAccountController.java]: Small typo... Thank you for being attentive.

**Review by volodymyrvynnyk — APPROVED** [review]

**Pull request merged**